### ISSUE TITLE
Border Count > 255 results in error: "BitsPerKey 16 exceeds maximum width 8" on CPU

### ISSUE THREAD

**Issue by rstokes92** [catboost/catboost#1132]
Problem: Using the `CatboostRegressor` If I set either `border_count` or `per_float_feature_quantization` to anything greater than 255 I get the following error
```
uncaught exception:
    address -> 0x10694ec00
    what() -> "catboost/private/libs/algo/index_hash_calcer.h:82: Internal CatBoost Error (contact 
    developers for assistance): BitsPerKey 16 exceeds maximum width 8"
    type -> TCatBoostException

```
My parameters are 
```
       CatBoostRegressor(
        random_seed=42,
        logging_level="Silent",
        nan_mode="Forbidden",
        loss_function="MAE",
        has_time=True,
        n_estimators=1500,
        depth=8,
        feature_border_type="Uniform",
        border_count=256,
    )
```
I want to improve the quantization of a particularly predictive feature as described in the parameter tuning [docs](https://catboost.ai/docs/concepts/parameter-tuning.html#golden-features)

catboost version: 0.20.2 (also got same with 0.17.5)
Operating System: Ubuntu 18.04
CPU: Running on AWS r3.2xlarge
Related Issue: Internal CatBoost Error: BitsPerKey 16 exceeds maximum width 8 #1010

**Comment by kizill** [comment]
We are currently working on this bug, thank you for information!

**Comment by kizill** [comment]
Fixed in master in e7c685a5dbc25b8c81522d1887afe422a6fc55f3. Will appear in next release, thank you for finding that bug 👍
